src/util/psbt/serialize.rs

```diff
@@ -83,12 +88,35 @@ impl Serialize for PublicKey {
 }
 
 impl Deserialize for PublicKey {
+    #[inline]
     fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
         PublicKey::from_slice(bytes)
             .map_err(|_| encode::Error::ParseFailed("invalid public key"))
     }
 }
 
+impl Serialize for PartialSig {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = self.signature.serialize_der().to_vec();
+        buf.push(self.sighash_type as u8);
+        buf
+    }
+}
+
+impl Deserialize for PartialSig {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        if bytes.len() <= 1 {
```

Comment:
Why is this checking `<= 1` ant the one above `is_empty`? I think they should be the same (presumably `is_empty` but see above).

If I'm not mistaken `from_str` could just call `deserialize()` after hex-decoding which would avoid such discrepancies.

Reply:
This is not a bug but rather tradeoff on how to report error.

If we have `len == 1`, we assume the last byte to be a sighash - and signature to be empty. Above, we error on it while parsing signature (empty signature will always error on parsing). Here, we explicitly check and error on empty sigs before parsing.

After consideration I think it is better to use error from signature deserializer and propagate it (i.e. they way you suggested to do above with `split_last`.

I am refactoring this PR and will use this strategy. Pls let me know if you do not agree.